core/src/scene/sceneLoader.cpp

```diff
@@ -66,6 +73,233 @@ glm::vec3 parseVec3(const Node& node) {
+    }
+
+    matTex.tex = sharedTexs[name];
+
+    if (mappingNode) {
+        std::string mapping = mappingNode.as<std::string>();
+        if (mapping == "uv") { matTex.mapping = MappingType::uv; }
+        else if (mapping == "planar") { logMsg("WARNING: planar texture mapping not yet implemented\n"); } // TODO
+        else if (mapping == "triplanar") { logMsg("WARNING: triplanar texture mapping not yet implemented\n"); } // TODO
+        else if (mapping == "spheremap") { matTex.mapping = MappingType::spheremap; }
+        else { logMsg("WARNING: unrecognized texture mapping \"%s\"\n", mapping.c_str()); }
+    }
+
+    return matTex;
+}
+
+void SceneLoader::loadTextures(YAML::Node textures, Scene& scene) {
+
+    if (!textures) {
+        return;
+    }
+
+    for (auto textureNode : textures) {
+
+        std::string name = textureNode.first.as<std::string>();
+        Node textureConfig = textureNode.second;
+
+        std::string file;
+        TextureOptions options = {GL_RGBA, GL_RGBA, {GL_LINEAR, GL_LINEAR}, {GL_CLAMP_TO_EDGE, GL_CLAMP_TO_EDGE} };
+
+        Node url = textureConfig["url"];
+        if (url) {
+            file = url.as<std::string>();
+        } else {
+            logMsg("WARNING: No url specified for texture \"%s\", skipping.\n", name.c_str());
+            continue;
+        }
+
+        Node filtering = textureConfig["filtering"];
+        if (filtering) {
+            std::string f = filtering.as<std::string>();
+            if (f == "linear") { options.m_filtering = { GL_LINEAR, GL_LINEAR }; }
+            else if (f == "mipmap") { options.m_filtering = { GL_LINEAR_MIPMAP_LINEAR, GL_LINEAR_MIPMAP_LINEAR }; }
+            else if (f == "nearest") { options.m_filtering = { GL_NEAREST, GL_NEAREST }; }
+        }
+
+        Node sprites = textureConfig["sprites"];
+        if (sprites) { logMsg("WARNING: sprite mapping not yet implemented\n"); } // TODO
+
+        scene.getTextures().emplace(name, std::make_shared<Texture>(file, options));
+    }
+}
+
+void SceneLoader::loadStyles(YAML::Node styles, Scene& scene) {
+
+    // Instantiate built-in styles
+    scene.getStyles().emplace_back(new PolygonStyle("polygons"));
+    scene.getStyles().emplace_back(new PolylineStyle("lines"));
+    scene.getStyles().emplace_back(new TextStyle("FiraSans", "text", 15.0f, 0xF7F0E1, true, true));
+    scene.getStyles().emplace_back(new DebugTextStyle("FiraSans", "debugtext", 30.0f, 0xDC3522, true));
+    scene.getStyles().emplace_back(new DebugStyle("debug"));
+
+    if (!styles) {
+        return;
+    }
+
+    for (auto styleIt : styles) {
+
+        Style* style = nullptr;
+
+        std::string styleName = styleIt.first.as<std::string>();
+        Node styleNode = styleIt.second;
+
+        Node baseNode = styleNode["base"];
+        if (baseNode) {
+            std::string baseString = baseNode.as<std::string>();
+            if (baseString == "lines") { style = new PolylineStyle(styleName); }
```

Comment:
Before adding a "named" style make sure same named style does not exists.

example: if a stylesheet defines a style named "lines" using "lines" base style..